### PR TITLE
pgo show backup Command Update

### LINKED ISSUE #1927: Backups can't be listed when pimrary is not in ready state
**Describe the bug**
Backups from a pgbackrest repository can't be listed while primary pod is not in ready state.

**To Reproduce**
1. Create cluster with local pgbackrest repository
2. Scale down primary node (in our case it crashed and could not be recovered)
3. ```pgo -n [ns] show backup [clustername]``` returns `Error: primary pod is not in Ready state`

**Expected behavior**
Backups can be listed, even if primary pod is not in ready state

**Please tell us about your environment:**

* Operating System: Centos
* Where is this running ( Local, Cloud Provider): Local
* Storage being used (NFS, Hostpath, Gluster, etc): Hostpath
* Container Image Tag: 4.4.1
* PostgreSQL Version: 12.1
* Platform (Docker, Kubernetes, OpenShift): K8S
* Platform Version: 1.17.9

As always: Thanks for great product and the support! :+1:

### PR DESCRIPTION
**Pull request by tjmoore4**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

Currently, the 'pgo show backup' functionality requires the
Primary pod to be in a Ready state. This is not technically required
because the relevant command is executed within the 'backrest-repo'
pod. 

**What is the new behavior (if this is a feature change)?**

This PR separates out the check for the Primary pod
from the 'show backup' functionality so that the backup information
can be shown regardless of the Primary pod's status.

**Other information**:
[ch9402]
fixes #1927 

For Testing:

Create a cluster
`pgo create cluster test1`
Once ready, exec in and kill an essential process to cause the pod to enter an unready state
`kubectl exec -it test1-559c4df8bf-dj42l -- bash`
While it the pod is recovering, run
`pgo backup test1`
and
` pgo show backup test1`
Until Ready, the backup command should display
"Error: primary pod is not in Ready state"
while the show backup command should give the requested information.

### CODE REVIEW

File: internal/apiserver/backrestservice/backrestimpl.go [hunk 1]
```diff
@@ -191,9 +191,19 @@ func CreateBackup(request *msgs.CreateBackrestBackupRequest, ns, pgouser string)
...
+			log.Error(err)
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = err.Error()
+			return resp
+		}
+
+		// check if primary is ready
+		err = isPrimaryReady(cluster, ns)
 
 		if err != nil {
```
Comment (jkatz) [inline internal/apiserver/backrestservice/backrestimpl.go:208]: Do we still need this check? If so, I'd recommend:

```suggestion
		// check if primary is ready
		if err := isPrimaryReady(cluster, ns); err != nil {
```

If not, please just drop it.

Comment (tjmoore4) [reply internal/apiserver/backrestservice/backrestimpl.go:208]: We do still want to do that check before taking the backup, but I will update it to the suggested change. Thanks!

Comment (jkatz) [reply internal/apiserver/backrestservice/backrestimpl.go]: Yes, for taking the backup we need that check. For showing backups ,we don't need that check.

**Review by jkatz — APPROVED** [review]

**Comment by jkatz** [comment]
@tjmoore4 Please rebase.

**Comment by tjmoore4** [comment]
@jkatz PR is rebased.

**Pull request merged**